Release 1.0.0 (dev branch) (#52)
nf-core/funcprofiler: docs/output.md

The directories listed below will be created in the results directory after the pipeline has finished. All paths are relative to the top-level results directory.

## Pipeline overview

> The pipeline processes data using the following steps:

Reviewer: [![suggestion(if-minor)](https://img.shields.io/badge/suggestion-if--minor-14B8A6?labelColor=3B82F6)](https://pullpo.io/cc?l=suggestion&d=if-minor) 
```suggestion
The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes data using the following steps:
```

I would recommend not modifying/removing boilerplate as far as possible, as this will become problematic in the future for you during template syncs (more conflicts to resolve).

- [FastQC](#fastqc) - Raw read QC and preprocessing
- [HUMANn v3 / v4](#humann-v3--v4) - Functional profiling via MetaPhlAn + HUMANn
- [FMH FunProfiler](#fmh-funprofiler) - Sketch-based functional profiling
- [mifaser](#mifaser) - Read-level functional profiling
- [DIAMOND blastx](#diamond-blastx) - Translated alignment against a protein database
- [EggNOG-mapper](#eggnog-mapper) - Functional annotation via orthology assignment
- [RGI BWT](#rgi-bwt) - Antimicrobial resistance gene identification
- [MultiQC](#multiqc) - Aggregate report describing results and QC from the whole pipeline
- [Pipeline information](#pipeline-information) - Report metrics generated during the workflow execution

> ## Other than FastQC, MultiQC and pipeline information, all other steps (the profilers) are off by default, and must be switched on manually.

Reviewer: [![nitpick](https://img.shields.io/badge/nitpick-F59E0B)](https://pullpo.io/cc?l=nitpick) 
```suggestion
:::note
Other than FastQC, MultiQC and pipeline information, all other steps (the profilers) are off by default, and must be switched on manually.
:::
```

Don't use titles for warning messages, but rather use the website admonitions: https://nf-co.re/docs/developing/documentation/website-markdown#notes

### FastQC

> 

Reviewer: [![nitpick(non-blocking)](https://img.shields.io/badge/nitpick-non--blocking-9CA3AF?labelColor=F59E0B)](https://pullpo.io/cc?l=nitpick&d=non-blocking) 
Our convention is to have a one-sentence description of each tool with a link to the tools homepage/github page before pipeline specific output documentation, I would recommend adding these back to bring back in line with the rest of the nf-core pipelines

Reviewer: This applies for everything below

- [Short reads QC and preprocessing](https://nf-co.re/subworkflows/fastq_shortreads_preprocess_qc/), see [Output section](https://nf-co.re/subworkflows/fastq_shortreads_preprocess_qc/#output) for details.
- Long reads QC and preprocessing (work in progress)

> ---

Reviewer: [![question(if-minor)](https://img.shields.io/badge/question-if--minor-14B8A6?labelColor=8B5CF6)](https://pullpo.io/cc?l=question&d=if-minor) 
What is the purpose of the `---` separator? The titles should act as sufficient section separators to me.

Reviewer: [![nitpick(blocking)](https://img.shields.io/badge/nitpick-blocking-374151?labelColor=F59E0B)](https://pullpo.io/cc?l=nitpick&d=blocking) 
And where are the FASTQC output files?

We want to keep the pipeline documentation self-contained as these are bundled (and archived) with teh pipeline code, please don't point to external subworkflow/module pages as these may change as versions get updated.

### HUMANn v3 / v4

> Enabled with `--run_humann_v3` or `--run_humann_v4`. Each sample is first run through MetaPhlAn to generate a taxonomic profile, which guides HUMANn functional profiling.

Reviewer 2: What is humann_v3 and v4? I would like to have a short 1,2 sentence summary of the underlying concept such that one understands the following output description

> <details markdown="1">

Reviewer 2: not sure how well the drop-down renders on nf-core website, but I would suggest to remove them. It's good to directly see the output structure of all tools

<summary>Output files</summary>

- `humann_v3/<sample>/` or `humann_v4/<sample>/`
  - `*_genefamilies.tsv`: Gene family abundances in reads per kilobase (RPK), stratified by contributing species.
  - `*_pathabundance.tsv`: Metabolic pathway abundances in RPK, stratified by species contribution.
  - `*_pathcoverage.tsv`: Pathway coverage scores (0–1), indicating the fraction of reactions detected per pathway.
- `metaphlan/<sample>/`
  - `*_profile.txt`: Species-level taxonomic abundance profile used as input to HUMANn.

</details>

---

### FMH FunProfiler

> Enabled with `--run_fmhfunprofiler`. Uses FracMinHash sketching to rapidly assign reads to functional categories.

Reviewer 2: Elaborate a bit

<details markdown="1">
<summary>Output files</summary>

- `fmhfunprofiler/<sample>/`
  - `*.ko.txt`: KO (KEGG Orthology) abundance table for the sample.

</details>

---

### mifaser

Enabled with `--run_mifaser`. Maps reads to functional databases at the protein level to produce enzyme function profiles.

<details markdown="1">
<summary>Output files</summary>

- `mifaser/<db_name>/<sample>/`
  - `analysis.tsv`: Tab-separated table of functional assignments with read counts per enzyme function (EC number).
  - `analysis.log`: Log file with run statistics including number of reads processed and assigned.

</details>

---

### DIAMOND blastx

> Enabled with `--run_diamond`. Performs fast translated alignment of metagenomic reads against a protein reference database. Each read is aligned in all six reading frames and only significant hits are reported.

Reviewer 2: ```suggestion
Enabled with `--run_diamond`. Performs fast translated alignment of (meta)genomic reads against a protein reference database. Each read is aligned in all six reading frames and only significant hits are reported.
```
